classes/i18n/LocaleFile.inc.php

```diff
@@ -82,34 +47,30 @@ function getFilename() {
 	 * @return string
 	 */
 	function translate($key, $params = array(), $locale = null) {
-		if ($this->isValid()) {
-			$key = trim($key);
-			if (empty($key)) {
-				return '';
-			}
+		if (!$this->isValid()) return null;
+		$key = trim($key);
+		if ($key === '') return '';
+
+		$pool = new Stash\Pool(Core::getStashDriver());
+		$item = $pool->getItem('locale-' . md5($this->getFilename()));
+		if ($item->isMiss() || filemtime($this->filename) >= $item->getCreation()->getTimestamp()) {
+			$item->set(LocaleFile::load($this->filename));
+			$pool->save($item);
+		}
+		$messages = $item->get();
```

Comment:
Just a check: does this still play nicely with the CustomLocale plugin?

Reply:
Yes, this is a straight-up replacement for the previous caching toolset we were using. I started looking into normal Gettext caching strategies but they tend to depend on `.mo` file compilation and I didn't think it necessary to bite that off as the same time as this change.